src/key-value.spec.js

```diff
@@ -0,0 +1,23 @@
+describe('ReactKeyValue', () => {
+  it('should have the proper class name');
+  it('should set the default props');
+  it('should get the initial state from the props');
+  it('should handle adding new rows');
+  it('should handle when a key changes');
+  it('should handle when a value changes');
+  it('should handle removing a row');
+  it('should render a KeyItem');
+  it('should render a ValueItem');
+  it('should have a renderRows method');
+  it('should have a renderAddButton method');
+
+  describe('react-key-value-rows', () => {
```

Comment:
Hmm I don't really get it. I wouldn't nest describes here, I think it's unnecessary. "react-key-value-rows" is the class name of an element inside the render method. It doesn't call the `renderRows` method. We just put the render logic of the rows into a method called renderRows to increase maintainability and readability. But this is just an implementation detail again which could be refactored in the future, doesn't matter. What matters is whether the rows are rendered properly based on `props.rows`.